src/query/interpreter.cpp

```diff
@@ -1909,6 +1918,119 @@ PreparedQuery PrepareSettingQuery(ParsedQuery parsed_query, const bool in_explic
+                                 TypedValue(std::to_string(interpreter->GetTransactionId())),
+                                 TypedValue(queries_summaries_typed)});
+            }
+          }
+        });
+        return results;
+      };
+      break;
+    }
+    case TransactionQueueQuery::Action::TERMINATE_TRANSACTIONS: {
+      std::vector<std::string> transaction_ids;
+      std::transform(transaction_query->transaction_id_list_.begin(), transaction_query->transaction_id_list_.end(),
+                     std::back_inserter(transaction_ids), [&evaluator](Expression *expression) {
+                       return std::string(expression->Accept(evaluator).ValueString());
+                     });
+      callback.header = {"transaction_id", "status"};
+      callback.fn = [interpreter_context, db_accessor, transaction_ids, username, isAdmin]() mutable {
+        spdlog::info("Number of transactions to kill in callback: {}", transaction_ids.size());
+        std::vector<std::vector<TypedValue>> results;
+        for (const std::string &transaction_id : transaction_ids) {
+          std::string status{"Transaction not found"};
+          interpreter_context->interpreters.WithLock(
+              [db_accessor, &transaction_id, &status, username, isAdmin](const auto &interpreters) {
+                for (Interpreter *interpreter : interpreters) {
+                  std::string intr_trans_str = std::to_string(interpreter->GetTransactionId());
+                  if (interpreter->HasActiveTransaction() && transaction_id == intr_trans_str) {
+                    if (interpreter->username_ == username || isAdmin) {
+                      // if (interpreter->in_explicit_transaction_) {
+                      //   interpreter->expect_rollback_ = true;
+                      // } else {
+                      interpreter->AbortTransactionByUser();
+                      interpreter->Abort();
+                      // }
+                      status = "Transaction killed";
+                    } else {
+                      status = "Not enough rights to kill the transaction";
+                    }
+                    break;
+                  }
+                }
+              });
+          results.push_back({TypedValue(transaction_id), TypedValue(status)});
+        }
+        return results;
+      };
+      break;
+    }
+  }
+
+  return callback;
+}
+
+PreparedQuery PrepareTransactionQueueQuery(ParsedQuery parsed_query, const std::optional<std::string> &username,
+                                           const uint64_t current_transaction_id, const bool in_explicit_transaction,
+                                           InterpreterContext *interpreter_context, DbAccessor *dba) {
+  if (in_explicit_transaction) {
+    // throw SettingConfigInMulticommandTxException{};
+  }
+
+  spdlog::info("Number of interpreters in prepare transactions: {}", interpreter_context->interpreters->size());
```

Comment:
people do not know what an interpreter is, either delete or put an appropriate message, possibly in debug mode